superset-frontend/src/explore/components/controls/ControlPopover/ControlPopover.tsx

```diff
@@ -89,24 +98,50 @@ const ControlPopover: React.FC<PopoverProps> = ({
 
   const handleOnVisibleChange = useCallback(
     (visible: boolean) => {
-      if (props.visible === undefined) {
+      if (visible === undefined) {
```

Comment:
Is this correct?
Actually, we will never `visible` state as `undefined`. 
I think this should be `visibleProp`?

Reply:
The local state is needed cause "visible" can be undefined, (leaving the component uncontrolled), so we can't just call the `onVisibleChange` and expect someone to update the state, the user might not be using this callback.